plotly/tools.py

```diff
@@ -51,6 +51,13 @@ def warning_on_one_line(message, category, filename, lineno,
 except ImportError:
     _numpy_imported = False
 
+try:
+    import scipy
+    import scipy.stats
+    _scipy_imported = True
+except ImportError:
+    _scipy_imported = False
+
```

Comment:
Note that this is going to be a merge conflict with #274 . I guess we'll need to have some specific requirements specified... How about a syntax like:

```
_scipy_imported
_scipy__spatial_imported
_scipy__cluster__hierarchy_imported
```

I'm not sure which is going to land first, but the latter will have to implement this.